packages/component/src/SendBox/ConnectivityStatus.js

```diff
@@ -1,110 +1,102 @@
-        </React.Fragment>
-      );
-    } else if ( status === 'reconnecting' ) {
-      return (
-        <React.Fragment>
-          <SpinnerAnimation />
-          {localize( 'INTERRUPTED_CONNECTION_NOTIFICATION', language )}
-        </React.Fragment>
-      );
-    } else if ( status === 'connectingslow' ) {
-      return (
-        <React.Fragment>
-          <WarningNotificationIcon />
-          {localize( 'SLOW_CONNECTION_NOTIFICATION', language )}
-        </React.Fragment>
-      );
-    } else if ( status === 'error' ) {
-      return (
-        <React.Fragment>
-          <ErrorNotificationIcon />
-          {localize( 'FAILED_CONNECTION_NOTIFICATION', language )}
-        </React.Fragment>
-      );
-    }
-  }
 
-  constructor( props ) {
-    super( props );
+class ConnectivityDebounce extends React.Component {
+  constructor(props) {
+    super(props);
 
     this.state = {
-      connectivityStatus: props.connectivityStatus,
-      connectingSlow: false,
-      language: props.language,
-      styleSet: props.styleSet,
-    }
+      renderConnectivity: props.children,
+      since: Date.now()
+    };
   }
 
-  componentDidUpdate( props ) {
-    const { connectivityStatus: { status } } = props;
-    const { connectingSlow } = this.state
+  componentWillReceiveProps(nextProps) {
+    const { children, debounce } = nextProps;
+
+    if (
+      children !== this.props.children
+      || debounce !== this.props.debounce
+    ) {
+      clearTimeout(this.timeout);
 
-    if ( status === 'connectingslow' && connectingSlow === false ) {
-      this.setState( () => ( { connectingSlow: true } ) );
+      this.timeout = setTimeout(() => {
+        this.setState(() => ({
+          renderConnectivity: children,
+          since: Date.now()
+        }));
+      }, Math.max(0, debounce - Date.now() + this.state.since));
     }
   }
 
-  static getDerivedStateFromProps( props, state ) {
-    if ( props !== state ) {
-      state = {
-        connectivityStatus: props.connectivityStatus,
-        connectingSlow: false,
-        language: props.language,
-        styleSet: props.styleSet,
-      }
-    }
-    return state;
+  componentWillUnmount() {
+    clearTimeout(this.timeout);
   }
 
   render() {
-    const { connectivityStatus, styleSet } = this.state;
-    const { status } = connectivityStatus;
-    return (
-      <div
-        aria-live="polite"
-        className={classnames( {
-          [ styleSet.errorNotification ]: ( status === 'error' ),
-          [ styleSet.warningNotification ]: status === 'connectingslow' || status === 'reconnecting',
-          [ styleSet.connectivityNotification ]: status === 'connected' || status === 'uninitialized' || status === 'reconnecting'
-        } )}
-        role="status"
-      >
-        {this.connectivityStatusText()}
-      </div>
-    );
+    const { renderConnectivity } = this.state;
+
+    return typeof renderConnectivity === 'function' ? renderConnectivity() : false;
```

Comment:
in a "Function as Children" pattern like what you have here, I typically just do something like below to self-document the intended control flow clearly without any naming sugar.

```javascript
  return typeof this.props.children === 'function' ? this.props.children() : false
```